src/test/java/io/tiledb/java/api/GroupTest.java

```diff
@@ -0,0 +1,211 @@
+    group.addMember("array3", false, "array3Name");
+
+    // reopen group
+    group.reopen(ctx, QueryType.TILEDB_READ);
+    Assert.assertEquals(3, group.getMemberCount());
+
+    // remove a member
+    group.reopen(ctx, TILEDB_WRITE);
+    group.removeMember("array2");
+
+    // check if member is removed
+    group.reopen(ctx, QueryType.TILEDB_READ);
+    Assert.assertEquals(2, group.getMemberCount());
+
+    group.close();
+  }
+
+  @Test
+  public void metadataTest() throws TileDBError {
+    Group.create(ctx, "test_group");
+    Group group = new Group(ctx, "test_group", TILEDB_WRITE);
+
+    NativeArray shortArray = new NativeArray(ctx, new short[] {18, 19, 20, 21}, Short.class);
+    NativeArray longArray = new NativeArray(ctx, new long[] {8L, 9L, 0L, 1L}, Long.class);
+    NativeArray intArray = new NativeArray(ctx, new int[] {7, 1, 3, 2}, Integer.class);
+
+    // put metadata in closed group
+    try {
+      group.putMetadata("key1", shortArray);
+      throw new TileDBError("Put metadata in closed group");
+    } catch (TileDBError error) {
+    }
+
+    // put metadata in open group
+    group.putMetadata("key1", shortArray);
+    group.putMetadata("key2", longArray);
+    group.putMetadata("key3", intArray);
+
+    // close group and reopen in read mode
+    Config config = new Config();
+    config.set("sm.group.timestamp_start", "10");
+    Context context = new Context(config);
+    group.reopen(context, TILEDB_READ);
+
+    // check if the correct metadata keys are present
+    Assert.assertEquals(BigInteger.valueOf(3), group.getMetadataNum());
+    Assert.assertTrue(group.hasMetadataKey("key1"));
+    Assert.assertTrue(group.hasMetadataKey("key2"));
+    Assert.assertTrue(group.hasMetadataKey("key3"));
+    Assert.assertFalse(group.hasMetadataKey("key4"));
+
+    // check values of metadata keys
+    NativeArray metadataShortArray = group.getMetadata("key1", Datatype.TILEDB_INT16);
+    NativeArray metadataLongArray = group.getMetadata("key2", Datatype.TILEDB_INT64);
+    NativeArray metadataIntArray = group.getMetadata("key3", Datatype.TILEDB_INT32);
+    Assert.assertNotNull(metadataShortArray);
+    Assert.assertNotNull(metadataIntArray);
+    Assert.assertNotNull(metadataLongArray);
+    Assert.assertArrayEquals(
+        (short[]) shortArray.toJavaArray(), (short[]) metadataShortArray.toJavaArray());
+    Assert.assertArrayEquals(
+        (long[]) longArray.toJavaArray(), (long[]) metadataLongArray.toJavaArray());
+    Assert.assertArrayEquals(
+        (int[]) intArray.toJavaArray(), (int[]) metadataIntArray.toJavaArray());
+
+    // close group and reopen in write mode
+    config = new Config();
+    config.set("sm.group.timestamp_start", "20");
+    context = new Context(config);
+    group.reopen(context, TILEDB_WRITE);
+    // delete a metadata entry
+    group.deleteMetadata("key2");
+    // TODO this delete functionality does not always work. The core library does not have any tests
```

Comment:
Let's make this a conditional failure when libtiledb version is less than 2.8.0 -- should be fixed in 2.8.1